src/analyses/goto_check.cpp

```diff
@@ -277,6 +278,23 @@ class goto_checkt
   bool enable_assumptions;
   bool enable_pointer_primitive_check;
 
+  /// Maps a named-check name to the corresponding boolean flag.
+  std::map<irep_idt, bool *> name_to_flag{
+    {"no-enum-check", &no_enum_check},
```

Comment:
Can you please insert a commit at the beginning of your commit series that entirely removes `no_enum_check` from goto_check.cpp? It is always `false`. Seems like an unnecessary piece of code introduced in 85b90f304ba90ca83d4f1f3c78e28293c51c057b, probably had a meaning at some point while developing that change back then.

Reply:
This flag is an override to the user-controlled `enable_enum_check` that is set to true before checking LHS in assignments:

```
      // Reset the no_enum_check with the flag reset for exception
      // safety
      {
        flag_resett resetter(i);
        resetter.set_flag(no_enum_check, true, "no_enum_check");
        check(assign_lhs);
      }
```

If we remove this flag CBMC will start checking LHS expressions, I think the problem could be that these assertions will trigger when the LHS is uninitialised and can take any integer value.